kore/src/Kore/Reachability/Claim.hs

```diff
@@ -530,8 +530,11 @@ checkImplicationWorker (ClaimPattern.refreshExistentials -> claimPattern) =
     -- the claim is fully simplified at every step, that should not be a
     -- problem.
     sideCondition =
-        SideCondition.assumeTrueCondition
-            (Condition.fromPredicate . Condition.toPredicate $ leftCondition)
+        SideCondition.fromConditionWithReplacements
+            ( Condition.fromPredicate
+                . Condition.toPredicate
```

Comment:
What does this do?

Reply:
It moves the `substitution` back into the `predicate`, I think the TODO above explains why we have to do this.

Reply:
Oh, I missed that note in GitHub. Thanks!